### ISSUE TITLE
Add to Gift Registry from Wishlist adding incorrect quantity in EE 2.3.1

### ISSUE THREAD

**Issue by dhiraj-danode** [magento/magento2#24048]
### Preconditions (*)

1.  Enterprise 2.3.1
2.

### Steps to reproduce (*)

1. Login to Account
2. Go to Wishlist
3. Add To Gift Registry
4. Go to Gift Registry
5. Check the quantity, incorrect quantity get added.
6. We added log module-gift-registry/Controller/Index/Wishlist.php, controller get called multiple times.
7. Please note this issue observed if only multiple wishlist is enabled. If its diabled its working fine

### Expected result (*)

1. [Screenshots, logs or description]
2. Correct Quantity should get added in Gift Registry

### Actual result (*)

1. [Screenshots, logs or description]
2. Added video for actual steps

**Comment by dhiraj-danode** [comment]
Below is video link to for actual steps

**Comment by engcom-Bravo** [comment]
Hello @dhiraj-danode. Thanks for applying. It is most likely You use some third-party extension for multiple wishlist, which is not available in a vanilla Enterprise Edition from the box, am I right?

**Comment by dhiraj-danode** [comment]
Hello @engcom-Bravo 

We are not using any third party extension. Multiple wish list available in Enterprise version.

**Comment by sudheers-kensium** [comment]
Hello @dhiraj-danode. Not reproduced in Magento EE 2.3.1

**Comment by dhiraj-danode** [comment]
Hello @sudheers-kensium 

Did you check when multiple wish list enable.
Because its only replicating when multiple wish list enable.

**Comment by engcom-Bravo** [comment]
Sorry @dhiraj-danode, this functionality is really already available in Enterprise version and seems Your issue is present. We need some additional recheck so we'll be back soon.

**Comment by magento-engcom-team** [comment]
@engcom-Bravo Thank you for verifying the issue.

Unfortunately, not enough information was provided to acknowledge ticket. Please consider adding the following:
- [ ] Add `"Reproduced on "` label(s) to this ticket based on verification result

Once all required information is added, please add label `"Issue: Confirmed"` again.
Thanks!

**Comment by sudheers-kensium** [comment]
> Hello @sudheers-kensium
> 
> Did you check when multiple wish list enable.
> Because its only replicating when multiple wish list enable.
> 
> Below is the video, its fresh magenot EE 2.3.1 version
> https://drive.google.com/file/d/1gh7RZPTUYqsCPfYZz2NmykhQZr07cymN/view

@dhiraj-danode Yes, I enabled the multiple wish list and verified in Magento EE 2.3.1 but not able to reproduce the issue.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Bravo 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-19022` were created

**Issue Available**: @engcom-Bravo, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by sdzhepa** [comment]
Hello @dhiraj-danode @engcom-Bravo 

Thank you for contribution and collaboration!

I moved this issue into Commerce Related private repository where it could be fixed and delivered.

[Current repository](https://github.com/magento/magento2) and [issue tracker](https://github.com/magento/magento2/issues) aimed at Magento Open Source version only and the main focus is community contribution/collaboration. It described in [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines) and it is a part of the issue report template:
> Verify, that the issue you are about to report does not relate to the Magento Commerce. GitHub is intended for Magento Open Source users to report on issues related to Open Source only. There are no account management services associated with GitHub. You can report Commerce-related issues in one of two ways:
> - You can use the Support portal associated with your account
> - If you are a Partner reporting on behalf of a merchant, use the Partner portal.
